lab2.js

```diff
@@ -58,15 +76,25 @@ var hoursSpentInDowington; // TODO: assign me the value of the
 // town, and the starting consumption rate, and returns the number
 // of hours the blob needs to ooze its way through that town.
 
-function hoursToOoze(population, peoplePerHour) {
+Blob.prototype.hoursToOoze = function(population, peoplePerHour) {
   // TODO: implement me based on the instructions above.
   // Be sure to then assign me to the Blob's prototype.
-}
+  var peopleEaten = 0;
+  var hoursEating = 0;
+  while (peopleEaten < population) {
+    peopleEaten += peoplePerHour;
+    peoplePerHour++;
+    hoursEating++;
+  }
+  return hoursEating;
+};
 
 assert(blob.hoursToOoze(0, 1) === 0, 'no people means no time needed.');
 assert(blob.hoursToOoze(1000, 1) === hoursSpentInDowington,
   'hoursSpentInDowington should match hoursToOoze\'s result for 1000');
-
+assert(blob.hoursToOoze(1000, 1000) === 1, 'should be able to eat everything in first hour');
+assert(blob.hoursToOoze(3, 1) === 2, '1 first hour, then 2 second hour');
+assert(blob.hoursToOoze(1000, 0) === hoursSpentInDowington + 1, 'should only set back one hour');
```

Comment:
I like that this and the third assert are testing similar things - you'll be able to quickly spot if things are off.